### DISCUSSION TITLE
Where can I find demos source code?

### DISCUSSION THREAD

**Discussion by me21** [WolfgangFahl/nicegui_widgets#63] [General]
Thank you for providing a number of advanced NiceGUI demos. I can browse them at https://ngdemo.bitplan.com/, but how do I access their source code to reuse in my software? I'm not sure, but the repository at https://github.com/WolfgangFahl/nicegui_widgets seems to contain only new components' source code without demos.

**Comment by WolfgangFahl** [comment]
Most demos have a pypi and or github link. Just click on e.g. 
<img width="97" alt="grafik" src="https://github.com/WolfgangFahl/nicegui_widgets/assets/1336221/87dc2f15-ae4f-4a25-8e82-a6b3b3efdc47"> and you'll end up in the correspdonding github repository.

The source code for the nicegui widgets demo site itself is at https://github.com/WolfgangFahl/nicegui_widgets/blob/main/ngwidgets/widgets_demo.py.  It has the split into the webserver https://github.com/WolfgangFahl/nicegui_widgets/blob/08086db0ce787c60c8858ffcc240fd0ccb7b5257/ngwidgets/widgets_demo.py#L525 with the routes to the pages and the per client solution at https://github.com/WolfgangFahl/nicegui_widgets/blob/08086db0ce787c60c8858ffcc240fd0ccb7b5257/ngwidgets/widgets_demo.py#L40
